HW2Question2Updatded.cpp

```diff
@@ -0,0 +1,90 @@
+//
+//  HW2Question2.cpp
+//  Codu
+//
+//  Created by Farida Eid on 5/27/17.
+//  Copyright © 2017 Farida Eid. All rights reserved.
+//
+
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+struct treeNode
+{
+    int value;
+    treeNode *left;
+    treeNode *right;
+    
+    treeNode(int v):value(v), left(nullptr), right(nullptr){}
+    
+};
+
+// Checks if node it is the tree or not
+
+bool checkNodeInTree(treeNode* root, int key)
```

Comment:
Having this check separated from the lowestCommonAncestor is kind of error prone. Ideally, there is one lowestCommonAncestor called by users, that calls checkNodeInTree and then calls an internal version of lowestCommonAncestor that is not public and can assume nodes are there.